src/js/parsers/section.js

```diff
@@ -80,31 +80,16 @@ class SectionParser {
     if (!finished) {
       let childNodes = isTextNode(element) ? [element] : element.childNodes;
 
-      if (this.state.section.isListSection) {
-        this.parseListItems(childNodes);
-      } else {
-        forEach(childNodes, el => {
-          this.parseNode(el);
-        });
-      }
+      forEach(childNodes, el => {
+        this.parseNode(el);
+      });
     }
 
     this._closeCurrentSection();
 
     return this.sections;
   }
 
-  parseListItems(childNodes) {
-    let { state } = this;
-    forEach(childNodes, el => {
-      let parsed = new this.constructor(this.builder).parse(el);
```

Comment:
Creating a new SectionParser instance was the main reason for getting rid of this code. To be able to detect nested lists we need access to the current state which is not available in the new instance.